services/QuillLMS/client/app/bundles/Teacher/components/assignment_flow/manage_units/unit.tsx

```diff
@@ -0,0 +1,229 @@
+    attribute: 'activity',
+    width: '646px',
+    isSortable: false,
+    noTooltip: true,
+    rowSectionClassName: 'lesson-activity',
+    headerClassName: 'lesson-activity-header'
+  },
+  {
+    name: 'Launch Lesson',
+    attribute: 'launchLesson',
+    width: '150px',
+    isSortable: false,
+    noTooltip: true,
+    rowSectionClassName: 'launch-lesson',
+    headerClassName: 'launch-lesson-header'
+  },
+  {
+    name: 'Customize',
+    attribute: 'customize',
+    width: '92px',
+    isSortable: false,
+    noTooltip: true
+  },
+  {
+    name: 'Download',
+    attribute: 'download',
+    width: '46px',
+    isSortable: false,
+    noTooltip: true
+  },
+]
+
+const sharedButtonClassName = "quill-button contained extra-small green focus-on-light"
+
+export const Unit = ({ data }) => {
+  const {
+    classroomActivities,
+    classroom_activities,
+    classrooms,
+    name,
+    num_students_assigned,
+    studentCount,
+    unit,
+    unitId,
+    unitName
+  } = data
+  const [lessonActivityId, setLessonActivityId] = React.useState<number>(null)
+  const [showModal, setShowModal] = React.useState<boolean>(false)
+
+  function assignedToText () {
+    // ensure classrooms is always an array as sometimes it is passed a set
+    // and we need to do a number of things with it that are better with an array
+    const classroomsArray = Array.isArray(classrooms) ? classrooms : Array.from(classrooms);
+    const studentCountData = num_students_assigned || studentCount;
+    return (
+      <div className="assigned-to">
+        <p>Assigned to</p>
+        <span className="bolded-value">{`${studentCountData} ${Pluralize('Student', studentCountData)}`}</span>
+        <p>in</p>
+        <span className="bolded-value">{`${classroomsArray.length} ${Pluralize('class', classroomsArray.length)} (${classroomsArray.join(', ')})`}</span>
+      </div>
+    )
+  };
+
+  function lockedInfo () {
+    const firstCa = classroomActivities.values().next().value;
+    const ownedByCurrentUser = firstCa?.ownedByCurrentUser;
+    if(ownedByCurrentUser) { return }
+    return(
+      <span className="locked-info">
+        <img alt="" src={`${EXTRA_SMALL_ICON_BASE_SRC}/locked.svg`} />
+        Created By {firstCa.ownerName}
+      </span>
+    );
+  };
+
+  function handleDataTableChipClick(e, id) {
+    setLessonActivityId(id)
+    handleToggleModal()
+  }
+
+  function handleToggleModal() {
+    setShowModal(!showModal)
+  }
+
+  function getClassroomActivities() {
+    if (Object.keys(classroomActivities)) {
+      return classroomActivities
+    }
+    if (Object.keys(classroom_activities)) {
+      return classroom_activities
+    }
+  }
+
+  function getActivityId(lessonActivity) {
+    const { activityId, activityUid, activity } = lessonActivity
+    return activityId || activityUid || activity?.uid
+  }
+
+  function getClassroomUnitId(lessonActivity) {
+    const { cuId, classroom_unit_id } = lessonActivity
+    return cuId || classroom_unit_id
+  }
```

Comment:
Same here. If it's just too complicated to deal with for this project that's fine, but might be worth noting in the PR description.